### PR TITLE
Remove unused packages

### PR DESCRIPTION
**Pull request by 8n8**
Tidied up a package.yaml file by removing some unused dependencies.

### CODE REVIEW

**Comment by avh4** [comment]
Thanks!  I'll get to this soon (and before the 0.8.7 release).

I'll also add `-Wunused-packages` to all the other package.yaml's in the project -- feel free to add that yourself if you get to it before I do.

**Comment by avh4** [comment]
I think I'm also gonna upgrade this to `-Werror=unused-packages` (which can be downgraded again in `cabal.project.local` if desired to speed up local development, but will help me remember to actually notice the warning when committing)

File: elm-format-lib/package.yaml [hunk 1]
```diff
@@ -23,24 +23,18 @@ default-extensions:
...
-  ansi-wl-pprint: ">= 0.6.9 && < 0.7"
-  array: ">= 0.5.4.0 && < 0.6"
   base: ">= 4.15.0.0 && < 5"
   bimap: ">= 0.4.0 && < 0.6"
   binary: ">= 0.8.9.0 && < 0.9"
   bytestring: ">= 0.11.3.0 && < 0.12"
   containers: ">= 0.6.5.1 && < 0.7"
-  directory: ">= 1.3.6.2 && < 2"
-  filepath: ">= 1.4.2.1 && < 2"
-  ghc-prim: ">= 0.5.2"
```
Comment (avh4) [inline elm-format-lib/package.yaml:40]: @8n8 Do you remember why you had to re-add ghc-prim in your initial PR?  It made me remove it again when I made the warning into an error, and it compiled fine for me locally.  I guess I'll see if CI fails.

Comment (8n8) [reply elm-format-lib/package.yaml:40]: Yes, I removed it when I saw the warning, but then when I built again from scratch it failed. I think I had to do `git clean -f` to remove the build caches to make it fail.

Comment (8n8) [reply elm-format-lib/package.yaml:40]: If you replace `import GHC.Prim` with `import GHC.Exts` in `elm-format-lib/src/Data/Utf8.hs` it fixes the issue.

The [docs](https://hackage.haskell.org/package/ghc-prim) for GHC.Prim say:

> This package contains the primitive types and operations supplied by GHC. It is an internal package, only for the use of GHC developers. GHC users should not use it! If you do use it then expect breaking changes at any time without warning. You should prefer to import GHC.Exts from the base package instead.

Comment (avh4) [reply elm-format-lib/package.yaml:40]: Ah, great!  I'll make the change to `GHC.Exts`.  I also reported [#23212: -Werror=unused-packages cannot be satisfied for ghc-prim · Issues · Glasgow Haskell Compiler / GHC · GitLab](https://gitlab.haskell.org/ghc/ghc/-/issues/23212).

**Comment by avh4** [comment]
Thanks for the cleanup!

File: elm-format-lib/package.yaml (hunk 1 above)
Comment (avh4) [reply elm-format-lib/package.yaml:40]: oh, also `dev/build.sh clean && dev/build.sh` was enough to trigger the original problem without having to re-clone.

Comment (8n8) [reply elm-format-lib/package.yaml:40]: Oh right. That's easier, thanks.

**Pull request merged**